### PR TITLE
chore(lint): graduate sonarjs/different-types-comparison warn → error (#943-1)

### LINKED ISSUE #943: chore(lint): graduate sonarjs type-checked rules warn → error
## 背景

#926 で \`parserOptions.projectService\` を有効化した際、SonarJS 由来の type-checked ルールが多数 wake up した。1PR で全部修正するには規模が広いので、warn に降格してこの issue で graduation を tracking する。

## 対象ルール (warn 中)

| ルール | hits | 内容 |
|---|---|---|
| \`sonarjs/no-alphabetical-sort\` | 40 | \`.sort()\` のデフォルト挙動への警告 |
| \`sonarjs/different-types-comparison\` | 19 | 型が異なる値の \`===\` / \`!==\` (常に false/true) |
| \`sonarjs/no-misleading-array-reverse\` | 11 | \`arr.reverse()\` を破壊的でないと誤解 |
| \`sonarjs/prefer-regexp-exec\` | 10 | \`str.match(re)\` より \`re.exec(str)\` 推奨 |
| \`sonarjs/no-undefined-argument\` | 5 | 不要な \`undefined\` 引数 |
| \`sonarjs/function-return-type\` | 5 | 戻り型が一致しない関数 |
| \`sonarjs/no-redundant-optional\` | 3 | \`undefined | T\` と \`?\` の重複 |
| \`sonarjs/no-selector-parameter\` | 1 | 引数で挙動切替 (split into 2 funcs) |
| \`sonarjs/deprecation\` | 1 | deprecated symbol 使用 |

## 戦略

各ルールを個別 PR で:
1. 違反箇所を triage (本物の bug か / 意図的か)
2. 修正 or rationale 付き \`eslint-disable-next-line\`
3. ルールを \`warn → error\` に格上げ

ルールごとに hits が小さい (≤40) ので、1 ルール = 1 PR で進めれば現実的。

\`different-types-comparison\` は **常に false/true となる比較を検出する** ルールで、本物の bug を含む可能性が高い (e.g., 2件は \`SettingsMcpTab.vue\` と \`presentMulmoScript/View.vue\` で型不一致 \`===\`)。これだけ先にやる価値あり。

## 完了条件

- [ ] 上記9ルールを \`error\` 化
- [ ] 各 PR で違反の triage 結果を description に明記

## 関連

- Roadmap: #927 / #926 (type-checked 有効化 PR)
- ベース: https://github.com/SonarSource/eslint-plugin-sonarjs

### PR DESCRIPTION
**Pull request by isamu**
## Summary

First cleanup PR off of #943's roadmap. Triages every `sonarjs/different-types-comparison` warn from #941 and graduates the rule to error. **Zero real bugs found** — every hit was a case where TS's static narrowing is stricter than the runtime, and the existing checks are correct defensive code.

Also narrows `parserOptions.projectService` to `{src,server,test,e2e}` (excludes `packages/` workspaces) to bound lint memory cost without losing test/e2e coverage of type-checked rules.

**Stacked on #941** (the type-checked-mode enable PR). Base will auto-redirect to `main` once #941 merges.

## Items to Confirm / Review

- [ ] **No behavior changes** — every fix preserves the runtime check that the rule complained about. The form changes (`!= null`, `?? ""`, `in` operator, `unknown[]` cast, optional body fields) match the original semantics.
- [ ] **Untrusted Express body shape change** — `RenderBeatBody.beatIndex: number` → `beatIndex?: number` etc. This makes the type honest about JSON parsing returning whatever the client sent. Routes still validate at runtime; downstream callers were already checking.
- [ ] **`DailyJsonShape.items: NewsItem[]` → `unknown[]`** in `server/workspace/news/reader.ts`. The downstream `.filter((item): item is NewsItem => …)` was already doing real validation work; pre-typing as `NewsItem[]` made it look dead. Filter body adjusted to narrow `unknown` → `Record<string, unknown>` first.
- [ ] **`projectService` scope narrowing** — `**/*.{ts,tsx}` → `{src,server,test,e2e}/**/*.{ts,tsx}`. `packages/` workspaces have separate tsconfigs and the type-info graph cost wasn't worth it for them. Test/e2e stays in scope per project preference.
- [ ] **Per-line disables (3 sites)** in cases where TS narrowing genuinely can't track the runtime semantics:
  - `src/plugins/presentMulmoScript/View.vue:901` — reactive ref re-read after a mutation
  - `test/agent/test_agent_config.ts:94`, `test/chat-index/test_indexer.ts:47,57` — array out-of-bounds (no `noUncheckedIndexedAccess` in tsconfig)

## User Prompt

> さてこのまま直せるものは直していこう。つぎのPRとして。  
> その変更、副作用ない？できるだけテストも考慮して直してね  
> ちょっと941, 現実的にバグになる可能性のたかいruleにしぼる？  
> #943-1で、941をそこにとりこんでからすすめよう

### Categorisation of all 19 original hits

| Pattern | Count | Fix |
|---|---|---|
| A. `JSON.stringify` lib type lies | 1 | `?? ""` fallback |
| B. Untrusted Express body required-fields | 5 | Make body field types optional + `!= null` idiom |
| C. Record indexed-access lies | 4 | `in` operator |
| D. JSON cast lies about structure | 1 | Type as `unknown[]`, narrow inside filter |
| E. `RegExpExecArray.index` no longer optional | 2 | Remove dead check |
| F. Array out-of-bounds | 4 | Remove dead `.filter(undefined)` (1) / per-line disable (3 in test) |
| G. Reactive ref re-read | 1 | Per-line disable |
| Already covered by other fixes earlier | 1 | (count off-by-one in initial 19→18 reconciliation) |

### Plan reference

Tracker: #927. This is the first cleanup PR for #943's SonarJS type-checked rules. Future PRs in #943: graduate `no-misleading-array-reverse` (11 hits) and `deprecation` (1 hit). #942 separately tracks `no-floating-promises` graduation.

## Test plan

- [x] `yarn format` clean
- [x] `yarn lint` 0 errors, 4527 warnings (down from 4534, 7 different-types-comparison errors resolved)
- [x] `yarn build:client` clean
- [x] `tsx --test` 3367 / 3367 pass

Closes #943 partially (different-types-comparison portion)
🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Promote the `sonarjs/different-types-comparison` ESLint rule to an error and adjust code and configs so existing defensive runtime checks remain valid without changing behavior.

Bug Fixes:
- Align Express request body types and JSON-derived shapes with their actual untrusted runtime inputs so defensive validation logic type-checks cleanly.
- Simplify and harden various runtime checks (e.g. record key presence, JSON stringify result, regex exec results, chart document inspection) without altering observable behavior.

Enhancements:
- Refine type annotations for untrusted request/JSON payloads and internal utilities to better reflect real runtime shapes and avoid misleading type guarantees.
- Replace fragile equality/undefined checks with safer patterns such as `!= null`, `in`-operator key checks, and more direct object/array handling.
- Remove dead filtering and optionality checks where runtime states cannot occur, and document intentional edge cases where TypeScript narrowing does not match runtime semantics.
- Narrow the scope of type-checked ESLint parser services to core `src`, `server`, `test`, and `e2e` paths to reduce linting memory and time costs while preserving coverage.

Tests:
- Add targeted inline ESLint rule disables in a few test and reactive-code sites where array bounds and ref re-reads intentionally rely on runtime behavior not captured by TypeScript’s type system.

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -116,13 +116,13 @@ export default [
     },
   },
   {
-    // Type-checked rules need tsc-backed parser services. Scope this
-    // block to files that are in a tsconfig — excludes `.mjs`/`.cjs`/
-    // `.cts` which are not part of any project's `include` array. Vue
-    // SFCs get parserOptions set in their own block at the bottom.
-    // Type-checked rules need tsc-backed parser services. Vue SFCs
+    // Type-checked rules need tsc-backed parser services. Scope to
```
Comment (isamu) [inline eslint.config.mjs:119]: 日本語コメントです。`src/server/test/e2e` に絞ると `scripts/` 配下の TypeScript が type-checked lint から外れます。`packages/` だけを避けたいなら、`scripts` は残す方が安全です。意図した除外か確認したいです。

File: test/agent/test_agent_config.ts [hunk 2]
```diff
@@ -86,8 +86,11 @@ describe("buildCliArgs", () => {
     });
     const pIdx = args.indexOf("-p");
     // `-p` is followed by either another flag or end-of-args, never
-    // by a plain text message.
+    // by a plain text message. Out-of-bounds array access returns
+    // undefined at runtime even though TS types `args[i]` as
```
Comment (isamu) [inline test/agent/test_agent_config.ts:90]: 日本語コメントです。`args.indexOf("-p")` が `-1` になった場合、`args[pIdx + 1]` が `args[0]` を読むので、テストが別の理由で通る可能性があります。`assert.ok(pIdx !== -1)` を追加するか、`string | undefined` を返す helper にすると意図が明確になります。

**Review by isamu — COMMENTED** [review]
日本語で2点コメントします。

1. `eslint.config.mjs` の `projectService` 対象を `src/server/test/e2e` に絞ると、`scripts/` 配下の TypeScript が type-checked lint の対象から外れます。`scripts/` には実際に TS ファイルがあるので、ここは意図した除外か確認したいです。もし `packages/` だけを外したいなら、`scripts` は残す方が安全です。

2. `test/agent/test_agent_config.ts` の `args.indexOf("-p")` は `-1` になり得ます。現状だと `args[pIdx + 1]` が `args[0]` を読んでしまうので、`-p` が消えたときにテストが別の理由で通る可能性があります。`assert.ok(pIdx !== -1)` を入れるか、`string | undefined` を返す helper にすると、テストの意図が明確になります。
